doc/global-json-response.md

```diff
@@ -0,0 +1,97 @@
+Global JSON Response
+===================
+part of [Qcode Documentation](index.md)
+
+* * *
+
+The global JSON response represents a response to be delivered to the client as JSON. It consists of three main elements:
+
+* Record
+* Messages
+* Action
+
+### Record
+The record consists of elements that repesent the record item. Each element represents the result of validation against the data model and has 4 properties:
+
+`Name` The name of the element.
+
+`Valid` The result of validation of the element against the data model.
+
+`Value` The value of the element.
+
+`Message` A message relating to the element to feed back to the client. E.g. That the element is invalid or why it is invalid.
+
+
+### Messages
+Messages allow for sending the client valuable information. There are 3 types of message with each having a single property of `value`. Only one message of each type may appear in the response.
+
+`Alert` A warning or alert.
+
+`Error` Indicate what went wrong.
+
+`Notify` A notification.
+
+
+### Action
+Actions suggest what the client should do next. Each action has a single property of `value` that is a URL. There is currently just one action available:
+
+`Redirect` Go to the given URL.
+
+
+Setting Up The Response
+-----------------------
+
+Details on setting up the response can be found in the [API].
+
+Usage
+-----
+
+### [qc::validate2model]
+This procedure validates input against the data model and modifies the `record` object of the response accordingly.
+
+### Custom Validation Handlers
```

Comment:
I actually started somewhat briefly on doing this but stopped to evaluate.

Currently, the registration processes are only for the blog project so adding the registration documentation here doesn't make sense at the moment. I considered moving the `register` and `validate` procs over to the library with this commit but then realised that there's also more stuff that needs moving over and thought it might perhaps be best done one pull request at a time?

Here's what I've established so far that still has to be moved over from blog to the library:

* new handler registration process
* `qc::handlers` API
* new filters
* new `conn_marshal` and the related request handlers
* `cast`, `castable`, and `is` ensembles as well as the `ensemble_handler`
* file management

Should I add the registration process and documentation into this pull request or in a later one?